### PR TITLE
api server-scheduler integration tests

### PR DESCRIPTION
**Pull request by pradyutsarma**

### CODE REVIEW

File: test/integration/api/api.test.js [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+		fakePolicy=JSON.parse(fs.readFileSync(__dirname+'/fakePolicy.json','utf8'));
+		async.series([startScheduler,startApiServer], function(err){
+			done(err);
+		});
+	});
+	
+    after(function(done) {
+    	try {
+        	scheduler.kill();
+        	apiServer.kill();
```
Comment (fraenkel) [inline test/integration/api/api.test.js]: if scheduler.kill fails, don't we still want to kill the api server?

File: test/integration/api/api.test.js [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+	context('Create Policy with schedules ',function(){
+		// Cleanup the DB before running every context
+		before(function(done){
+			var deletePolicyOptions={
+					url:apiServerURI+'/v1/policies/dummy',
+					method:'DELETE',
+					json:true
+			};
+			request(deletePolicyOptions,function(error,response,body){
+				done();
```
Comment (fraenkel) [inline test/integration/api/api.test.js]: I would assume there are 2 conditions we expect and the rest are failures?!?

File: test/integration/api/api.test.js [hunk 3]
```diff
@@ -0,0 +1,467 @@
...
+				var getSchedulesOptions={
+						url:schedulerURI+'/v2/schedules/dummy',
+						method:'GET',
+						json:true
+				};
+				request(getSchedulesOptions,function(error,response,body){
+					expect(error).to.be.null;
+					expect(response.statusCode).to.equal(200);
+					expect(response.body.schedules.specific_date).to.not.be.empty;
+					expect(response.body.schedules.recurring_schedule).to.not.be.empty;
```
Comment (fraenkel) [inline test/integration/api/api.test.js]: isn't this redundant with the next check?

File: test/integration/api/api.test.js [hunk 4]
```diff
@@ -0,0 +1,467 @@
...
+					method:'DELETE',
+					json:true
+			};
+			request(deletePolicyOptions,function(error,response,body){
+				done();
+			});		
+		});	
+
+		it('should create the policy only',function(done){
+			var newFakePolicy=JSON.parse(JSON.stringify(fakePolicy));
```
Comment (fraenkel) [inline test/integration/api/api.test.js]: can we use more meaningful names, like policyNoSchedules?

File: test/integration/api/api.test.js [hunk 5]
```diff
@@ -0,0 +1,467 @@
...
+				});
+			});		
+		});
+
+		it('Should update a policy and associated schedules',function(done){
+			var newFakePolicy=JSON.parse(JSON.stringify(fakePolicy));
+			newFakePolicy.schedules.recurring_schedule[0].instance_max_count=8;
+			async.series([	
+			function updateDummyPolicy(callback){
+				var updateOptions={
```
Comment (fraenkel) [inline test/integration/api/api.test.js]: This doesn't update a policy.

File: test/integration/api/api.test.js [hunk 6]
```diff
@@ -0,0 +1,467 @@
...
+					expect(response.body.schedules.specific_date).to.have.lengthOf(2);
+					callback(error,response);
+				});
+			}
+			],done);
+		});
+
+	});
+	
+	context('Failure scenarios',function(){
```
Comment (fraenkel) [inline test/integration/api/api.test.js]: This is not a context.

File: test/integration/api/api.test.js [hunk 7]
```diff
@@ -0,0 +1,467 @@
...
+					url:apiServerURI+'/v1/policies/dummy',
+					method:'DELETE',
+					json:true
+			};
+			request(deletePolicyOptions,function(error,response,body){
+				done();
+			});		
+		});	
+
+		it('Should fail to create policy and associated schedules due to some validation error with the policy',function(done){
```
Comment (fraenkel) [inline test/integration/api/api.test.js]: This should be under Create policy context as a new context called Validation.

File: test/integration/api/api.test.js [hunk 8]
```diff
@@ -0,0 +1,467 @@
...
+				request(getSchedulesOptions,function(error,response,body){
+					expect(error).to.be.null;
+					expect(response.statusCode).to.equal(404);
+					callback(error,response);
+				});
+			}
+			],done);
+		});	
+		
+		it('Should return a 404 status code while deleting a non-existing policy and associated schedules',function(done){
```
Comment (fraenkel) [inline test/integration/api/api.test.js]: Part of With no policies

File: test/integration/api/api.test.js [hunk 9]
```diff
@@ -0,0 +1,467 @@
...
+				};
+				request(deletePolicyOptions,function(error,response,body){
+					expect(response.statusCode).to.equal(404);
+					expect(response.body.success).to.equal(false);
+					expect(response.body.error.statusCode).to.equal(404);
+					done();
+				});
+		});	
+		
+		it('Should return a 404 status code while retrieving a non-existing policy',function(done){
```
Comment (fraenkel) [inline test/integration/api/api.test.js]: Same as above.

File: test/integration/api/api.test.js (hunk 3 above)
Comment (pradyutsarma) [inline test/integration/api/api.test.js]: Fixed

File: test/integration/api/api.test.js (hunk 4 above)
Comment (pradyutsarma) [inline test/integration/api/api.test.js]: Done

File: test/integration/api/api.test.js (hunk 5 above)
Comment (pradyutsarma) [inline test/integration/api/api.test.js]: Policy updated too before triggering the update call

File: test/integration/api/api.test.js (hunk 6 above)
Comment (pradyutsarma) [inline test/integration/api/api.test.js]: Done

File: test/integration/api/api.test.js (hunk 7 above)
Comment (pradyutsarma) [inline test/integration/api/api.test.js]: Refactored to support this

File: test/integration/api/api.test.js (hunk 8 above)
Comment (pradyutsarma) [inline test/integration/api/api.test.js]: Done

File: test/integration/api/api.test.js (hunk 9 above)
Comment (pradyutsarma) [inline test/integration/api/api.test.js]: Didn't add explicit tests for this. Already tested as part of the existing tests.

**Comment by fraenkel** [comment]
The indentation is inconsistent.

**Comment by fraenkel** [comment]
Merged your previous verison and fixed up the indentation

**Pull request closed without merging**